nav2_behavior_tree/plugins/decorator/planner_selector_node.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Pablo Iñigo Blasco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+
+#include "std_msgs/msg/string.hpp"
+#include "behaviortree_cpp_v3/decorator_node.h"
+
+#include "nav2_behavior_tree/plugins/decorator/planner_selector_node.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_behavior_tree
+{
+
+using std::placeholders::_1;
+
+PlannerSelector::PlannerSelector(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::DecoratorNode(name, conf)
+{
+  auto node = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+
+  last_selected_planner_.data = "";
+
+  std::string planner_selector_topic;
+
+  node->get_parameter_or<std::string>(
```

Comment:
If this wasn't declared, then it will never be able to `get` it. Also, I think this should be a port so that if you had multiple planner selectors, they could each have set their own individual topic (vs this would make a single param impact all of them)

Reply:
One of the possible solution could be adding a constant string-input port to declare explicitly the topic name of the planner selector in the bt-xml file.

Reply:
Yeah, so if we make an input port a string for topic name, with a logical default, then that would work and replace use of node parameters.

Reply:
I think this also means that the topic subscription must be deferred to the first tick call where we have access to the values input ports values. (not a big deal)

This opens also the question of a dynamic topic name (I would keep it simple and would just use the first tick update)

Reply:
https://github.com/ros-planning/navigation2/blob/main/nav2_behavior_tree/plugins/action/back_up_action.cpp#L30 we rely on getting blackboard variables in the constructor before a `tick` when the tree is constructed. So its not waiting until the first tick to access them, yeah? 

I think we can assume static topic names safely, every other BT node doesn't assume that changes.

Reply:
Sounds good.
I will do that in the way you say.